tracer/src/Datadog.Trace/Iast/ReturnedHeadersAnalyzer.cs

```diff
@@ -0,0 +1,160 @@
+// <copyright file="ReturnedHeadersAnalyzer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Specialized;
+using System.Net;
+using System.Net.Mime;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+#if !NETFRAMEWORK
+using Microsoft.AspNetCore.Http;
+#endif
+using static Datadog.Trace.Telemetry.Metrics.MetricTags;
+
+#nullable enable
+namespace Datadog.Trace.Iast;
+
+internal static class ReturnedHeadersAnalyzer
+{
+    private const string ContentTypeLow = "content-type";
+    private const string XContentTypeOptionsLow = "x-content-type-options";
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ReturnedHeadersAnalyzer));
+
+#if NETFRAMEWORK
+    internal static void Analyze(NameValueCollection headers, IntegrationId integrationId, string serviceName, int responseCode)
+    {
+        try
+        {
+            if (string.IsNullOrEmpty(serviceName) || IsIgnorableResponseCode((HttpStatusCode)responseCode))
+            {
+                return;
+            }
+
+            IastModule.OnExecutedSinkTelemetry(IastInstrumentedSinks.XContentTypeHeaderMissing);
+
+            string contentTypeValue = string.Empty;
+            string contentOptionValue = string.Empty;
+
+            // We iterate instead of trying to get the key directly because keys are case insensitive
+            foreach (var headerKey in headers.AllKeys)
```

Comment:
I'm confused, we _want_ a case insensitive comparison right? Because that's effectively what we're using in the loop (just in a slower, more allocatey way 😄)

Reply:
But I still don't understand why we need to iterate. We'd get the same behaviour if just accessed the header directly right?

Reply:
Yes, you are right. I have also refactored the code so we don't need a body for .net framework and another for netcore. Thanks!